server/src/test/java/io/crate/expression/scalar/arithmetic/TrigonometricFunctionsTest.java

```diff
@@ -76,24 +78,36 @@ public void testEvaluate() throws Exception {
     }
 
     @Test
-    public void testEvaluateAsinOnIllegalArgument() {
-        expectedException.expect(IllegalArgumentException.class);
-        expectedException.expectMessage("input value 2.0 is out of range. Values must be in range of [-1.0, 1.0]");
-        assertEvaluate("asin(2.0)", 0);
+    public void test_relationships() throws Exception {
+        // https://en.wikipedia.org/wiki/Inverse_trigonometric_functions#Relationships_between_trigonometric_functions_and_inverse_trigonometric_functions
+        Percentage percentage = Percentage.withPercentage(99.9);
+        assertEvaluate("sin(asin(0.3))", 0.3);
+        assertEvaluate("sin(acos(0.3))", Math.sqrt(1 - Math.pow(0.3, 2)));
+        assertEvaluate("sin(atan(0.3))",
+            (Double val) -> assertThat(val).isCloseTo(0.3 / Math.sqrt(1 + Math.pow(0.3, 2)), percentage));
+        assertEvaluate("cos(asin(0.3))", Math.sqrt(1 - Math.pow(0.3, 2)));
+        assertEvaluate("cos(acos(0.3))", (Double val) -> assertThat(val).isCloseTo(0.3, percentage));
     }
 
     @Test
-    public void testEvaluateAcosOnIllegalArgument() {
-        expectedException.expect(IllegalArgumentException.class);
-        expectedException.expectMessage("input value 2.0 is out of range. Values must be in range of [-1.0, 1.0]");
-        assertEvaluate("acos(2.0)", 0);
+    public void test_atan_values_greater_and_less_than_1() throws Exception {
```

Comment:
We can remove this completely no? if you think is neccessary just move them to testEvaluate()?

Reply:
I don't see anything wrong with having a dedicated test case for it. I'll keep it